lib/iris/fileformats/netcdf.py

```diff
@@ -54,6 +54,8 @@
 import iris.fileformats._pyke_rules
 import iris.io
 import iris.util
+from itertools import repeat
```

Comment:
@jswanljung We've adopted an import ordering within iris, which is basically builtins, third-party, then iris specific modules. 

This said, `from itertools` is a `built-in` import and belongs up after the `import collections` line.